scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/view/FilterView.java

```diff
@@ -162,26 +188,224 @@ public void clearCriteria() {
+        JSONArray constraintsArray = new JSONArray();
+        for (int i = 0; i < widgetCount; i++) {
+            RowFilter row = (RowFilter) filterPanel.getMember(i);
+            String field = row.fieldsList.getSelectedValue();
+            String action = row.actionList.getSelectedValue();
+            String value = row.getValue();
+
+            JSONObject constraintObject = new JSONObject();
+            constraintObject.put("field", new JSONString(field.toString()));
+            constraintObject.put("action", new JSONString(action.toString()));
+            constraintObject.put("value", new JSONString(value.toString()));
+
+            constraintsArray.set(i, constraintObject);
+        }
+        jsonObject.put("filterName", new JSONString(filterNameTextBox.getText()));
+        jsonObject.put("matchAny", new JSONString(matchAnyButton.getValue().toString()));
+        jsonObject.put("constraints", constraintsArray);
+
+        return jsonObject.toString();
+    }
+
+    public void exportFilterConfiguration() {
+        String jsonConfig = generateFilterJsonString();
+        Settings.get().setSetting("filterConfig", jsonConfig);
+
+        String url = GWT.getModuleBaseURL() + "exportfilter";
+        url += "?filterConfigJson=" + jsonConfig;
+        url += "&filterName=" + filterNameTextBox.getText();
+
+        Window.open(url, "_blank", "");
+    }
+
+    public void loadFilterConfiguration(String jsonContent) {
+        JSONObject jsonObject = null;
+
+        try {
+            JSONValue parsedJsonString = JSONParser.parseStrict(new HTML(jsonContent).getText());
+            jsonObject = parsedJsonString.isObject();
+
+        } catch (JSONException e) {
+            LogModel.getInstance().logCriticalMessage(e.getMessage());
+        }
+
+        filterPanel.removeMembers(filterPanel.getMembers());
+
+        JSONString filterName = jsonObject.get("filterName").isString();
+        JSONString matchAny = jsonObject.get("matchAny").isString();
+        JSONArray constraints = jsonObject.get("constraints").isArray();
+        for (int i = 0; i < constraints.size(); i++) {
+            JSONObject field = constraints.get(i).isObject();
+            RowFilter newRow = new RowFilter();
+            newRow.fieldsList.setSelectedIndex(getSelectedValueIndex(newRow.fieldsList,
+                                                                     field.get("field").isString().stringValue()));
+            newRow.setActionsAccordingToSelectedField();
+            newRow.actionList.setSelectedIndex(getSelectedValueIndex(newRow.actionList,
+                                                                     field.get("action").isString().stringValue()));
+            newRow.setValue(field.get("value").isString().stringValue());
+            filterPanel.addMember(newRow);
+            updateRemoveButtonStatus();
+        }
+        if (matchAny.isString().stringValue().equals("true")) {
```

Comment:
Is it possible for stringValue() to return null? If yes, I think it would be better to change the order of the condition ("true".equals(matchAny.isString().stringValue()) in order to avoid a NullPointerException

Reply:
ok